src/Adapter/Kpi/AverageOrderValueKpi.php

```diff
@@ -26,41 +26,73 @@
 
 namespace PrestaShop\PrestaShop\Adapter\Kpi;
 
-use ConfigurationKPI;
-use Context;
 use HelperKpi;
+use PrestaShop\PrestaShop\Adapter\LegacyContext;
+use PrestaShop\PrestaShop\Core\ConfigurationInterface;
 use PrestaShop\PrestaShop\Core\Kpi\KpiInterface;
+use PrestaShopBundle\Translation\TranslatorInterface;
 
 /**
  * @internal
  */
 final class AverageOrderValueKpi implements KpiInterface
 {
+    /**
+     * @var TranslatorInterface
+     */
+    private $translator;
+
+    /**
+     * @var ConfigurationInterface
+     */
+    private $configuration;
+
+    /**
+     * @var LegacyContext
+     */
+    private $contextAdapter;
+
+    /**
+     * @param TranslatorInterface $translator
+     * @param ConfigurationInterface $configuration
+     * @param LegacyContext $contextAdapter
+     */
+    public function __construct(
+        TranslatorInterface $translator,
+        ConfigurationInterface $configuration,
+        LegacyContext $contextAdapter
+    ) {
+        $this->translator = $translator;
+        $this->configuration = $configuration;
+        $this->contextAdapter = $contextAdapter;
+    }
+
     /**
      * {@inheritdoc}
      */
     public function render()
     {
-        $translator = Context::getContext()->getTranslator();
-
         $helper = new HelperKpi();
         $helper->id = 'box-average-order';
         $helper->icon = 'account_balance_wallet';
-        $helper->color = 'color1';
-        $helper->title = $translator->trans('Average Order Value', [], 'Admin.Global');
-        $helper->subtitle = $translator->trans('30 days', [], 'Admin.Global');
+        $helper->color = 'color4';
+        $helper->title = $this->translator->trans('Average Order Value', [], 'Admin.Global');
+        $helper->subtitle = $this->translator->trans('30 days', [], 'Admin.Global');
 
-        if (ConfigurationKPI::get('AVG_ORDER_VALUE') !== false) {
-            $helper->value = $translator->trans(
+        if ($this->configuration->get('AVG_ORDER_VALUE') !== false) {
+            $helper->value = $this->translator->trans(
                 '%amount% tax excl.',
-                ['%amount%' => ConfigurationKPI::get('AVG_ORDER_VALUE')],
+                ['%amount%' => $this->configuration->get('AVG_ORDER_VALUE')],
                 'Admin.Orderscustomers.Feature'
             );
         }
 
-        $helper->source = Context::getContext()->link->getAdminLink('AdminStats')
-            . '&ajax=1&action=getKpi&kpi=average_order_value';
-        $helper->refresh = (bool) (ConfigurationKPI::get('AVG_ORDER_VALUE_EXPIRE') < time());
+        $helper->source = $this->contextAdapter->getAdminLink('AdminStats', true, [
+            'ajax' => 1,
+            'action' => 'getKpi',
+            'kpi' => 'average_order_value',
+        ]);
+        $helper->refresh = (bool) ($this->configuration->get('AVG_ORDER_VALUE_EXPIRE') < time());
```

Comment:
```suggestion
        $helper->refresh = (bool) ($this->configuration->getBool('AVG_ORDER_VALUE_EXPIRE') < time());
```

Reply:
Hum, but `AVG_ORDER_VALUE_EXPIRE` contain a timestamp in fact and we need to retrieve this data as it is.
Is there something I'm missing?

Reply:
I think @FabienPapet missed the comparison `< time`. So in fact it could be configuration->getInt() to compare two integers instead of integer and string (because by default all configuration->get('whatever') will be fetched as string